Tests/Rules/AvoidUsingAllowUnencryptedAuthentication.tests.ps1

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+BeforeAll {
+    $settings = @{
+        IncludeRules = @('PSAvoidUsingAllowUnencryptedAuthentication')
+        Rules        = @{
+            PSAvoidUsingAllowUnencryptedAuthentication = @{
+                Enable = $true
+            }
+        }
+    }
+
+    $violationMessage = [regex]::Escape("The insecure AllowUsingUnencryptedAuthentication switch was used. This should be avoided except for compatability with legacy systems.")
+}
+
+Describe "AvoidUsingAllowUnencryptedAuthentication" {
+    Context "When there are violations" {
+        It "detects unencrypted authentication violations" {
+            (Invoke-ScriptAnalyzer -ScriptDefinition 'Invoke-WebRequest foo -AllowUnencryptedAuthentication' -Settings $settings).Count | Should -Be 1
+            (Invoke-ScriptAnalyzer -ScriptDefinition 'Invoke-RestMethod foo -AllowUnencryptedAuthentication' -Settings $settings).Count | Should -Be 1
+            (Invoke-ScriptAnalyzer -ScriptDefinition 'iwr foo -AllowUnencryptedAuthentication' -Settings $settings).Count | Should -Be 1
+        }
+
+        It "has the correct description message" {
```

Comment:
normally we don't assert against error messages because it can make maintenance hard but there is no code sharing here so still ok. I suggest adding an assertion though on the rule name because this way we know it is this rule and not another rule that got triggered.

Reply:
On second thought, I think a test for verifying the violation message should only exist for rules in which we generate dynamic errors, ex: https://github.com/PowerShell/PSScriptAnalyzer/blob/master/Tests/Rules/AvoidUsingBrokenHashAlgorithms.tests.ps1#L14

I've removed the test from this new rule, as I don't think it adds any true coverage or value.